### DISCUSSION TITLE
Can someone help with Systemd setup?

### DISCUSSION THREAD

**Discussion by millaw73** [Taxel/PlexTraktSync#683] [Q&A] [answered]
What am I doing something wrong? 

- I installed using the GitHub clone method and I'm trying to set it up to actively watch and scrobble automatically in the background. Any pointer? Running this on Debian 10.
- I tried with: ExecStart=plextraktsync watch without success before changing to: ExecStart=/home/jackboy/PlexTraktSync/plextraktsync watch
- Below is what my Systemd entry looks like.

[Unit]
Description=PlexTraktSync watch daemon
After=network-online.target

[Service]
ExecStart=/home/jackboy/PlexTraktSync/plextraktsync watch
Restart=on-failure
RestartSec=10
User=jackboy
Group=jackboy

[Install]
WantedBy=multi-user.target

**Comment by stevietv** [comment]
I think you need to do this:

ExecStart /usr/bin/python3 - m /path/to/plextraktsync watch

Havent tried it but that's be my guess

**Reply by millaw73** [reply]
Thanks a lot for the prompt response. I'll try that in a bit and provide feedback.

**Reply by glensc** [reply]
`/home/jackboy/PlexTraktSync/plextraktsync.sh` - use existing path to script. 

`ls -l /home/jackboy/PlexTraktSync/plextraktsync.sh` to test that script exists.

**Reply by glensc** [reply]
@stevietv `python3 -m` wants module name as an argument so that will not work. also, you need to change the directory to git checkout dir with git installation mode.

- https://www.python.org/dev/peps/pep-0338/

**Reply by millaw73** [reply]
I ran /home/jackboy/PlexTraktSync/plextraktsync.sh as directed and the result is below:
-rwxr-xr-x 1 jackboy jackboy 85 Jan  1 18:25 /home/jackboy/PlexTraktSync/plextraktsync.sh

- Does that mean I should use the following instead: ExecStart=/home/jackboy/PlexTraktSync/plextraktsync.sh watch or is there something else that must be added before "watch"?

Really appreciate the help.

**Reply by stevietv** [reply]
If you can run this `/home/jackboy/PlexTraktSync/plextraktsync.sh watch` and get a valid output then you likely need to do this: 

`ExecStart=/home/jackboy/PlexTraktSync/plextraktsync.sh watch`

**Comment by glensc** [comment, marked as answer]
interesting post I found how to run docker containers in systemd:
- https://blog.container-solutions.com/running-docker-containers-with-systemd

it's not my recommendation or answer on how to proceed but some may find this useful.

**Reply by millaw73** [reply]
I'm really grateful for all the help. In case anyone wants to implement this, but doesn't know how, here's how I did it below:

- sudo nano /etc/systemd/system/PlexTrackSync.service

Then copy the entry below and paste to your window and make changes where indicated:

[Unit]
Description=PlexTraktSync watch daemon
After=network-online.target

[Service]
ExecStart=/home/jackboy/PlexTraktSync/plextraktsync.sh watch (**change this to your install path**)
Restart=on-failure
RestartSec=10
User=jackboy (**use your username here**)
Group=jackboy (**use your username here**)

[Install]
WantedBy=multi-user.target

Save it with Ctrl + X and answer Y, then enter.
**Now do the following:**

Reload the service files to include the new service.

- sudo systemctl daemon-reload

Start service

- sudo systemctl start PlexTrackSync.service

Check status of service

- sudo systemctl status PlexTrackSync.service

Enable service to load on every reboot

- sudo systemctl enable PlexTrackSync.service

**Reply by Rocknrolldoggie** [reply]
Hi Millaw73

Was a bit nervous giving this a go but your post made it sound so simple & all went well except when checking status it constantly shows as "Activating" . Any ideas ?

Solved it: had to remove the .sh after plextraktsync.sh watch
